setup.iss

```diff
@@ -663,7 +681,14 @@ begin
 
         // Unzip
         WizardForm.StatusLabel.Caption := 'Unzipping Freelancer: HD Edition';
-        UnZip(ExpandConstant('{tmp}\freelancerhd.zip'),ExpandConstant('{app}'));
+        //UnZip(ExpandConstant('{tmp}\{#MyZipName}.zip'),ExpandConstant('{app}'));
+        UnZip7Zip(ExpandConstant('{tmp}\{#MyZipName}.zip'),ExpandConstant('{app}'));
+        UpdateProgress(90);
+
+        // Copy mod files
+        WizardForm.StatusLabel.Caption := 'Moving Freelancer: HD Edition';
+        DirectoryCopy(ExpandConstant('{app}\{#MyFolderName}'),ExpandConstant('{app}'));
```

Comment:
I understand we're copying the files back from a folder too deep because 7-zip is extracting it with a parent folder. I wonder if there is a command line arguement we can use to prevent this from happening?

Reply:
I looked around trying to find a way to copy only the contents of the folder without the parent, but I couldn't find any commands at all that successfully did it, at least for me. I'll look around some more. Here's what we want to do: https://stackoverflow.com/questions/34723163/extract-sub-folder-contents-without-parent-folder-using-7zip-in-batch-files

Reply:
If we have to extract with the full path, then a "move" rather than a "copy and delete" would be acceptable. My only worry is if we copy its temporarily using more space I'd rather not need to.

Reply:
Looks like the [RenameFile function](https://jrsoftware.org/ishelp/index.php?topic=isxfunc_renamefile) will fail if a file exists with the same name, and also [doesn't copy directories](https://stackoverflow.com/questions/27068381/inno-setup-renaming-a-directory-fails#comment42656418_27068381).

(Side note: apparently if folders don't already exist, both FileCopy and RenameFile [won't create folders](https://stackoverflow.com/a/66085856). I don't know if this would be a problem for FL HD, but something to keep in mind.)

Maybe something like iterating through all the files in the extracted FL HD folder, and if a file of the same name exists from the vanilla version that was copied to the HD folder, remove it, then replace it with the FL HD version. Unsure how easy that would be to do with Inno functions. Any thoughts?

Reply:
@keenanweaver I think I've found a solution that will be ok. I've modified the DirectoryCopy function to include a Boolean parameter called "Move". If this is true it will delete each file as it copies it, meaning the installer won't temporarily take up double the space. I think this is probably good enough.

Could you give me write access to your fork please? I'll add the commit and merge.

Thanks again for the pull request!